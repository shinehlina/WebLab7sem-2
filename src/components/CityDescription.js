import React from "react";
import { Col } from "react-materialize";
import Table from "react-materialize/lib/Table";

class CityDescription extends React.Component {
  render() {
    return (
      <Col m={6} s={12}>
        <Table className="grey" m={6} s={12}>
          <tbody>
            <tr>
              <td>Coordinates</td>
              <td className="right-align">
                [{this.props.longtitude} : {this.props.latitude}]
              </td>
            </tr>
            <tr>
              <td>Weather description</td>
              <td className="right-align">{this.props.description}</td>
            </tr>
            <tr>
              <td>Temperature</td>
              <td className="right-align">{this.props.temperature}K</td>
            </tr>
            <tr>
              <td>Humidity</td>
              <td className="right-align">{this.props.humidity}%</td>
            </tr>
            <tr>
              <td>Wind</td>
              <td className="right-align">{this.props.windSpeed}m/c</td>
            </tr>
            <tr>
              <td>Pressure</td>
              <td className="right-align">{this.props.pressure}</td>
            </tr>
          </tbody>
        </Table>
      </Col>
    );
  }
}

export default CityDescription;
