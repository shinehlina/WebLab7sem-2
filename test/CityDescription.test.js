import React from "react";
 import CityDescription from "../src/components/CityDescription";
 import { shallow } from './enzyme';
 import { shallowToJson } from "enzyme-to-json";

 it("CityDescription didn't changed", () => {
   const app = shallow(
     <CityDescription
       latitude={30}
       longtitude={60}
       description={"windy"}
       temperature={273}
       humidity={20}
       windSpeed={10}
       pressure={1000}
     />
   );
   expect(shallowToJson(app)).toMatchSnapshot();
 });