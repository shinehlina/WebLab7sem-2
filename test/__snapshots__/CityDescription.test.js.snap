// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityDescription didn't changed 1`] = `
<Table
  className=" grey lighten-1"
>
  <tbody>
    <tr
      style={
        Object {
          "borderColor": "white",
        }
      }
    >
      <td>
        Coordinates
      </td>
      <td
        className="right-align"
      >
        [60 : 30]
      </td>
    </tr>
    <tr
      style={
        Object {
          "borderColor": "white",
        }
      }
    >
      <td>
        Weather description
      </td>
      <td
        className="right-align"
      >
        windy
      </td>
    </tr>
    <tr
      style={
        Object {
          "borderColor": "white",
        }
      }
    >
      <td>
        Temperature
      </td>
      <td
        className="right-align"
      >
        273K
      </td>
    </tr>
    <tr
      style={
        Object {
          "borderColor": "white",
        }
      }
    >
      <td>
        Humidity
      </td>
      <td
        className="right-align"
      >
        20%
      </td>
    </tr>
    <tr
      style={
        Object {
          "borderColor": "white",
        }
      }
    >
      <td>
        Wind
      </td>
      <td
        className="right-align"
      >
        10m/c
      </td>
    </tr>
    <tr
      style={
        Object {
          "borderColor": "white",
        }
      }
    >
      <td>
        Pressure
      </td>
      <td
        className="right-align"
      >
        1000
      </td>
    </tr>
  </tbody>
</Table>
`;
